File: 성민/레벨2/괄호회전하기.js
function isBracket(index, s) {
  const stack = [];
  let i = index;
  let cnt = 0;
  while (cnt < s.length) {
    if (s[i] === "]") {
      if (stack[stack.length - 1] !== "[") return false;
      else stack.pop();
    } else if (s[i] === ")") {
      if (stack[stack.length - 1] !== "(") return false;
      else stack.pop();
    } else if (s[i] === "}") {
      if (stack[stack.length - 1] !== "{") return false;
      else stack.pop();
    } else {
      stack.push(s[i]);
    }
    cnt++;
    i = i + 1 === s.length ? 0 : i + 1;
  }
  return stack.length === 0 ? true : false;
}
function solution(s) {
  let answer = 0;
  const strLen = s.length;
  let i = 0;

  while (i < strLen) {
    if (isBracket(i, s)) answer++;
    i++;
  }
  return answer;
}

console.log(solution("[](){}"));

File: 종우/괄호회전하기.py
# 1005_3.py
# 괄호 회전하기
# https://school.programmers.co.kr/learn/courses/30/lessons/76502

def solution(s):
    answer = 0
    nrot = len(s)
    for i in range(nrot):
        if iscor(s):
            answer += 1
        s = lrot(s)
    return answer

def lrot(s):
    return s[1:] + s[0]

def iscor(s):
    open_list = ["[", "{", "("]
    close_list = ["]", "}", ")"]
    stack = []
    for i in s:
        if i in open_list:
            stack.append(i)
        elif i in close_list:
            if len(stack) < 1:
                return False
            pos = close_list.index(i)
            if open_list[pos] == stack[len(stack)-1]:
                stack.pop()
    if len(stack) == 0:
        return True
    return False
        
    